src/App.tsx

```diff
@@ -2,24 +1,44 @@
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [visiblePerPage, setVisiblePerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(0);
+  const pages: string[][] = Array.from(
+    { length: Math.ceil(items.length / visiblePerPage) },
+    (_, i) =>
+      items.slice(i * visiblePerPage, i * visiblePerPage + visiblePerPage),
+  );
```

Comment:
The logic for splitting items into pages is a bit complex and hard to read. Consider extracting this into a separate function for better readability and maintainability. This would also allow you to unit test this logic separately.